src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/CSharpInstantiateArgumentExceptionsCorrectly.Fixer.cs

```diff
@@ -13,5 +20,77 @@ namespace Microsoft.NetCore.CSharp.Analyzers.Runtime
+                        title: MicrosoftNetCoreAnalyzersResources.InstantiateArgumentExceptionsCorrectlyFlipArgumentOrder,
+                        createChangedDocument: c => SwapArgumentsOrder(context.Document, creation, paramPosition, creation.ArgumentList.Arguments.Count, c),
+                        equivalenceKey: MicrosoftNetCoreAnalyzersResources.InstantiateArgumentExceptionsCorrectlyFlipArgumentOrder);
+                }
+                context.RegisterCodeFix(codeAction, diagnostic);
+            }
+        }
+
+        private static async Task<Document> SwapArgumentsOrder(Document document, ObjectCreationExpressionSyntax creation, int paramPosition, int argumentCount, CancellationToken token)
+        {
+            DocumentEditor editor = await DocumentEditor.CreateAsync(document, token).ConfigureAwait(false);
+            SyntaxNode parameter = AddNameOfIfLiteral(creation.ArgumentList.Arguments[paramPosition].Expression, editor.Generator);
+            SyntaxNode newCreation;
+            if (argumentCount == 2)
+            {
+                if (paramPosition == 0)
+                {
+                    newCreation = editor.Generator.ObjectCreationExpression(creation.Type, creation.ArgumentList.Arguments[1], parameter);
+                }
+                else
+                {
+                    newCreation = editor.Generator.ObjectCreationExpression(creation.Type, parameter, creation.ArgumentList.Arguments[0]);
+                }
+            }
+            else // 3 arguments
+            {
+                if (paramPosition == 0)
+                {
+                    newCreation = editor.Generator.ObjectCreationExpression(creation.Type, creation.ArgumentList.Arguments[1], parameter, creation.ArgumentList.Arguments[2]);
+                }
+                else
+                {
+                    newCreation = editor.Generator.ObjectCreationExpression(creation.Type, parameter, creation.ArgumentList.Arguments[1], creation.ArgumentList.Arguments[0]);
+                }
+            }
+            editor.ReplaceNode(creation, newCreation);
+            return editor.GetChangedDocument();
+        }
+
+        private static async Task<Document> AddNullMessageToArgumentList(Document document, ObjectCreationExpressionSyntax creation, CancellationToken token)
+        {
+            DocumentEditor editor = await DocumentEditor.CreateAsync(document, token).ConfigureAwait(false);
+            SyntaxNode argument = AddNameOfIfLiteral(creation.ArgumentList.Arguments.First().Expression, editor.Generator);
+            SyntaxNode newCreation = editor.Generator.ObjectCreationExpression(creation.Type, editor.Generator.Argument(editor.Generator.NullLiteralExpression()), argument);
+            editor.ReplaceNode(creation, newCreation);
+            return editor.GetChangedDocument();
+        }
+
+        private static SyntaxNode AddNameOfIfLiteral(ExpressionSyntax expression, SyntaxGenerator generator)
```

Comment:
This should possibly be changed to `IOperation argumentValue` and the check below can be changed to `if (argumentValue is ILiteralOperation literal)`